kythe/go/util/riegeli/golden_test.go

```diff
@@ -19,24 +19,60 @@ package riegeli_test
 import (
 	"encoding/hex"
 	"io"
+	"io/ioutil"
 	"os"
+	"strings"
 	"testing"
 
 	"kythe.io/kythe/go/storage/stream"
 	"kythe.io/kythe/go/util/riegeli"
 
 	"github.com/golang/protobuf/proto"
+	"github.com/google/go-cmp/cmp"
 
 	spb "kythe.io/kythe/proto/storage_go_proto"
+	rmpb "kythe.io/third_party/riegeli/records_metadata_go_proto"
 )
 
 var (
-	goldenJSONFile = "testdata/golden.entries.json"
-
-	goldenUncompressedRiegeliFile = "testdata/golden.entries.uncompressed.riegeli"
-	goldenBrotliRiegeliFile       = "testdata/golden.entries.brotli.riegeli"
+	goldenJSONFile            = "testdata/golden.entries.json"
+	goldenMetadataFile        = "testdata/golden.records_metadata.textproto"
+	goldenRiegeliFilePrefix   = "testdata/golden.entries"
+	goldenRiegeliFileVariants = []string{
+		"uncompressed",
+		"uncompressed_transpose",
+		"brotli",
+		"brotli_transpose",
+	}
 )
 
+func BenchmarkGoldenTestData(b *testing.B) {
+	for _, variant := range goldenRiegeliFileVariants {
+		file := strings.Join([]string{goldenRiegeliFilePrefix, variant, "riegeli"}, ".")
+		b.Run(variant, func(b *testing.B) { benchGoldenData(b, file) })
+	}
+}
+
+func benchGoldenData(b *testing.B, goldenRiegeliFile string) {
```

Comment:
Can you please document what this benchmark is benchmarking, at least in broad terms?

Reply:
✓